pkg/modules/modules.go

```diff
@@ -10,70 +10,13 @@ SPDX-License-Identifier: Apache-2.0
 package modules
 
 import (
-	"crypto"
-	"fmt"
-	"github.com/filecoin-project/mir/pkg/clients"
-	"github.com/filecoin-project/mir/pkg/reqstore"
-	"github.com/filecoin-project/mir/pkg/timer"
+	t "github.com/filecoin-project/mir/pkg/types"
 )
 
-// The Modules structs groups the modules a Node consists of.
-type Modules struct {
-	Net           Net              // Sends messages produced by Mir through the network.
-	Hasher        Hasher           // Computes hashes of requests and other data.
-	Crypto        Crypto           // Performs cryptographic operations (except for computing hashes)
-	App           App              // Implements user application logic. The user is expected to provide this module.
-	WAL           WAL              // Implements a persistent write-ahead log for the case of crashes and restarts.
-	ClientTracker ClientTracker    // Keeps the state related to clients and validates submitted requests.
-	RequestStore  RequestStore     // Provides persistent storage for request data.
-	Protocol      Protocol         // Implements the logic of the distributed protocol.
-	Interceptor   EventInterceptor // Intercepts and logs all internal _Events_ for debugging purposes.
-	Timer         Timer            // Tracks real time (e.g. for timeouts) and injects events accordingly.
+// Module generalizes the ActiveModule and PassiveModule types.
+type Module interface {
+	ImplementsModule()
```

Comment:
If we do so, the method will be not exported and there will be no way for other packages to directly implement the interface.

Reply:
I just checked that it works with the lowercase version. Go uses duck typing and there is no way to prohibit a struct from implementing a method with this name. The visibility only determines who can call the method.

Reply:
Otherwise, unexported methods from different packages don't match, see [here](https://blog.chewxy.com/2018/03/18/golang-interfaces/#sealed-interfaces).

Reply:
I see. Well, then I want to take my suggestion back: let's keep the uppercase version :)